### PR TITLE
EVA-3478 - Validation runs only when necessary

### PR DESCRIPTION
**Pull request by tcezard**

### CODE REVIEW

**Review by apriltuesday — COMMENTED** [review]
Thanks for adding the README! :clap:

File: README.md [hunk 1]
```diff
@@ -1,2 +1,57 @@
...
+
+Example:
+```shell
+vcf,fasta,report
+/full/path/to/vcf_file1.vcf,/full/path/to/genome.fa,/full/path/to/genome_assembly_report.txt
+/full/path/to/vcf_file2.vcf,/full/path/to/genome.fa,/full/path/to/genome_assembly_report.txt
+/full/path/to/vcf_file3.vcf,/full/path/to/genome2.fa,/full/path/to/genome_assembly_report2.txt
+```
+
+### The metadata spreadsheet
```
[the remark this answers is not in the document]
Comment (tcezard) [reply README.md:31]: I've added a stub

File: bin/eva-sub-cli.py [hunk 2]
```diff
@@ -41,38 +24,17 @@ def get_vcf_files(mapping_file):
...
-                       help="Username used for connecting to the ENA webin account")
-    group.add_argument("--password",
-                       help="Password used for connecting to the ENA webin account")
+    argparser.add_argument("--username",
+                           help="Username used for connecting to the ENA webin account")
+    argparser.add_argument("--password",
+                           help="Password used for connecting to the ENA webin account")
+    argparser.add_argument("--resume", default=False, action='store_true',
+                           help="Resume the process execution from where it left of. This is only supported for a "
+                                "subset of the tasks")
```
[the remark this answers is not in the document]
Comment (tcezard) [reply bin/eva-sub-cli.py]: It's only the upload at the moment. The native nextflow will be able to resume as well but I don't think the docker/singularity will.

File: eva_sub_cli/main.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+from eva_sub_cli import SUB_CLI_CONFIG_FILE, __version__
+from eva_sub_cli.docker_validator import DockerValidator
+from eva_sub_cli.reporter import READY_FOR_SUBMISSION_TO_EVA
+from eva_sub_cli.submit import StudySubmitter
+
+VALIDATE = 'validate'
+SUBMIT = 'submit'
+
+
+logging_config.add_stdout_handler()
```
[the remark this answers is not in the document]
Comment (tcezard) [reply eva_sub_cli/main.py]: Removed

File: tests/test_main.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+                validator.create_reports.assert_called_once_with()
+                validator.update_config_with_validation_result.assert_called_once_with()
+
+            # Submit was created
+            m_submitter.assert_any_call(self.test_sub_dir, m_get_vcf.return_value, self.metadata_json,
+                                        submission_config=m_config.return_value)
+            with m_submitter() as submitter:
+                submitter.submit.assert_called_once_with(resume=False)
+
+    def test_orchestrate_validate_no_submit(self):
```
[the remark this answers is not in the document]
Comment (tcezard) [reply tests/test_main.py]: Good catch I changed it.

**Review by apriltuesday — APPROVED** [review]

**Review by nitin-ebi — APPROVED** [review]

**Pull request merged**